operators/src/engine/operator.rs

```diff
@@ -65,41 +64,80 @@ pub trait PlotOperator: CloneablePlotOperator + Send + Sync + std::fmt::Debug {
     }
 }
 
-pub type InitializedVectorOperator =
-    dyn InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>;
+pub trait InitializedRasterOperator: Send + Sync {
+    /// Get the result descriptor of the `Operator`
+    fn result_descriptor(&self) -> &RasterResultDescriptor;
 
-pub type InitializedRasterOperator =
-    dyn InitializedOperator<RasterResultDescriptor, TypedRasterQueryProcessor>;
+    /// Instantiate a `TypedVectorQueryProcessor` from a `RasterOperator`
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor>;
 
-pub type InitializedPlotOperator =
-    dyn InitializedOperator<PlotResultDescriptor, TypedPlotQueryProcessor>;
+    /// Wrap a box around a `RasterOperator`
+    fn boxed(self) -> Box<dyn InitializedRasterOperator>
+    where
+        Self: Sized + 'static,
+    {
+        Box::new(self)
+    }
+}
 
-pub trait InitializedOperator<Descriptor, Processor>: Send + Sync {
+pub trait InitializedVectorOperator: Send + Sync {
     /// Get the result descriptor of the `Operator`
-    fn result_descriptor(&self) -> &Descriptor;
+    fn result_descriptor(&self) -> &VectorResultDescriptor;
 
     /// Instantiate a `TypedVectorQueryProcessor` from a `RasterOperator`
-    fn query_processor(&self) -> Result<Processor>;
+    fn query_processor(&self) -> Result<TypedVectorQueryProcessor>;
 
     /// Wrap a box around a `RasterOperator`
-    fn boxed(self) -> Box<dyn InitializedOperator<Descriptor, Processor>>
+    fn boxed(self) -> Box<dyn InitializedVectorOperator>
     where
         Self: Sized + 'static,
     {
         Box::new(self)
     }
 }
 
-impl<R, Q> InitializedOperator<R, Q> for Box<dyn InitializedOperator<R, Q>>
-where
-    R: ResultDescriptor,
-    Q: QueryProcessor,
-{
-    fn result_descriptor(&self) -> &R {
+pub trait InitializedPlotOperator: Send + Sync {
```

Comment:
Why do we need a trait here instead of a concrete type?

Reply:
I didn't design it this way but just split up the type